core/src/main/java/org/apache/james/core/healthcheck/Result.java

```diff
@@ -26,62 +26,27 @@
 public class Result {
 
     public static Result healthy(ComponentName componentName) {
-        return Builder.builder().componentName(componentName).status(ResultStatus.HEALTHY).build();
+        return new Result(componentName, ResultStatus.HEALTHY, Optional.empty(), Optional.empty());
     }
 
     public static Result unhealthy(ComponentName componentName, String cause) {
-        return Builder.builder().componentName(componentName).status(ResultStatus.UNHEALTHY).cause(cause).build();
+        return new Result(componentName, ResultStatus.UNHEALTHY, Optional.of(cause), Optional.empty());
     }
 
     public static Result unhealthy(ComponentName componentName, String cause, Throwable error) {
-        return Builder.builder().componentName(componentName).status(ResultStatus.UNHEALTHY).cause(cause).error(error).build();
+        return new Result(componentName, ResultStatus.UNHEALTHY, Optional.of(cause), Optional.of(error));
```

Comment:
The builder version was more readable IMO

Reply:
I think this version is ok: from the outside we have nice factory methods with good names and internally we have some calls to the complex constructor.

Reply:
(I just find sad that we had the builder and removed it)